File: Java/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.Arrays;

class Solution {
    public int solution(int[] array) {

        int[] count = new int[1000]; // 원소가 0 ~999니까 

        for(int i = 0; i < array.length; i++) {
            count[array[i]]++; // count[i]번째에 해당하면 +1
        }
        
        int max = 0;
        int maxCount = 0; // 최빈값 체크
        int mode = 0;
        // 일단 각 항목별 등장횟수 체크 후
        for(int i = 0; i < count.length; i++){
            if(count[i] > max ) {
                max = count[i];
            }
        }
        // 최빈값 확인
        for(int i = 0; i < count.length; i++){
            if(count[i] == max ) {
                mode = i; // 최빈값 저장
                maxCount++;
            }
        }
        
        if(maxCount >= 2) {
            return -1;
        } else {
            return mode;
        }
    }
}

File: JavaScript/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    let count = {};

    for (let i = 0; i < array.length; i++) {
        count[array[i]] = (count[array[i]] || 0) + 1;
    }
    
    let max = Math.max(...Object.values(count));
    let same = Object.values(count).filter(v => v === max).length;

    for (let key in count) {
        if (count[key] === max) {
            console.log(count[key])
            return same > 1 ? -1: Number(key);
        }
    }
}